src/libs/SidebarUtils.js

```diff
@@ -85,22 +88,37 @@ function setIsSidebarLoadedReady() {
  * @returns {String[]} An array of reportIDs sorted in the proper order
  */
 function getOrderedReportIDs(currentReportId, allReportsDict, betas, policies, priorityMode, allReportActions) {
+    // Generate a unique cache key based on the function arguments
+    const cachedReports = JSON.stringify([currentReportId, allReportsDict, betas, policies, priorityMode, Object.values(allReportActions).length], (key, value) => {
```

Comment:
`allReportActions` is an object with a key value, which the key is `reportActions_###` and the value an array. 
Getting the Object.values just gives you the reports you have and not the messages. Can you try to use
```
allReportActions['reportActions_${currentReportId}']?.length ?: 1
```
Will be more accurate to check if the number of messages changed for that report